### PR TITLE
Add a simple mechanism to generate test binaries from sycl source files.

### PR DESCRIPTION
**Pull request by aarongreig**
Also implement tests for urProgramCreateWithIL entry point.

### CODE REVIEW

File: test/conformance/source/environment.cpp [hunk 1]
```diff
@@ -256,44 +256,49 @@ std::string
...
+            break;
+        }
+    }
 
-    uint32_t address_bits;
-    auto device = instance->GetDevices()[device_index];
-    if (uur::GetDeviceAddressBits(device, address_bits)) {
-        error = "failed getting device address bits supported";
+    if (binary_name.empty()) {
+        error = "couldn't find appropriate device binary";
```
Comment (martygrant) [inline test/conformance/source/environment.cpp:277]: Wonder if some of these hard-coded error messages could be standardised in one location or added to the UR_RESULT_X enums maybe?

**Review by martygrant — APPROVED** [review]
LGTM, just a couple simple queries.

**Review by kbenzie — CHANGES_REQUESTED** [review]

File: test/conformance/program/urProgramCreateWithIL.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+//
+#include <uur/fixtures.h>
+
+using urProgramCreateWithILTest = uur::urProgramILBinaryTest;
+
+UUR_INSTANTIATE_DEVICE_TEST_SUITE_P(urProgramCreateWithILTest);
+
+TEST_P(urProgramCreateWithILTest, Success) {
```
[the remark this answers is not in the document]
Comment (aarongreig) [reply test/conformance/program/urProgramCreateWithIL.cpp:10]: yeah you do, that'll come in the tests for other entry points

**Review by kbenzie — APPROVED** [review]
LGTM

**Review by veselypeta — APPROVED** [review]
LGTM

File: test/conformance/source/environment.cpp (hunk 1 above)
Comment (aarongreig) [reply test/conformance/source/environment.cpp]: alright they're now used properly and visible in test output

**Pull request merged**